src/test/java/org/broadinstitute/hellbender/tools/copynumber/gcnv/GermlineCNVIntervalVariantComposerUnitTest.java

```diff
@@ -0,0 +1,199 @@
+    private static final SimpleInterval TEST_INTERVAL = new SimpleInterval("1", 1, 10000);
+
+    /* sums up to unity in probability space */
+    private static final double[] TEST_VALID_LOG_POSTERIOR_VECTOR = new double[] {
+            -22.133276724683618,
+            -4.3825075766712871,
+            -0.012754265709645551,
+            -8.6265377789688955,
+            -21.918647174298602,
+            -22.133276474165779};
+
+    /* expected PLs, copy-number MAP call, and GQ for above */
+    private static final List<Integer> TEST_EXPECTED_PLS = new ArrayList<>(Arrays.asList(96, 18, 0, 37, 95, 96));
+    private static final int TEST_EXPECTED_MAP_CN = 2;
+    private static final int TEST_EXPECTED_GQ = 18;
+
+    /* does not sum up to unity in probability space */
+    private static final double[] TEST_INVALID_LOG_POSTERIOR_VECTOR = new double[] {
+            -22.133276724683618,
+            -4.3825075766712871,
+            -0.12754265709645551,
+            -8.6265377789688955,
+            -21.918647174298602,
+            -22.133276474165779};
+
+    private static final String TEST_SAMPLE_NAME = "TEST_SAMPLE_NAME";
+
+    private static final List<IntegerCopyNumberState> TEST_COPY_NUMBER_STATE_LIST = new ArrayList<>();
+    private static final IntegerCopyNumberStateCollection TEST_COPY_NUMBER_STATE_COLLECTION;
+    static {
+        IntStream.range(0, 6).forEach(cn -> TEST_COPY_NUMBER_STATE_LIST.add(new IntegerCopyNumberState(cn)));
+        TEST_COPY_NUMBER_STATE_COLLECTION = new IntegerCopyNumberStateCollection(TEST_COPY_NUMBER_STATE_LIST.stream()
+                .map(IntegerCopyNumberState::toString)
+                .collect(Collectors.toList()));
+    }
+
+    private static final CopyNumberPosteriorDistribution TEST_DISTRIBUTION = new CopyNumberPosteriorDistribution(
+            IntStream.range(0, TEST_VALID_LOG_POSTERIOR_VECTOR.length)
+                    .boxed()
+                    .collect(Collectors.toMap(IntegerCopyNumberState::new, cn -> TEST_VALID_LOG_POSTERIOR_VECTOR[cn])));
+
+    private static final LocatableCopyNumberPosteriorDistribution TEST_LOCATABLE_DISTRIBUTION =
+            new LocatableCopyNumberPosteriorDistribution(TEST_INTERVAL, TEST_DISTRIBUTION);
+
+    /**
+     * Assert that exception is thrown if normalized posterior probabilities do not add up to 1
+     */
+    @Test(expectedExceptions = IllegalArgumentException.class)
+    public void testCopyNumberPosteriorDistributionValidation() {
+        final CopyNumberPosteriorDistribution posteriorRecord = new CopyNumberPosteriorDistribution(
+                IntStream.range(0, TEST_VALID_LOG_POSTERIOR_VECTOR.length)
+                        .boxed()
+                        .collect(Collectors.toMap(IntegerCopyNumberState::new,
+                                cn -> TEST_INVALID_LOG_POSTERIOR_VECTOR[cn])));
+        final LocatableCopyNumberPosteriorDistribution locatablePosteriorRecord =
```

Comment:
The exception should already be thrown by now, no need for these 2 lines

Reply:
done.